include/instantiate.h

```diff
@@ -151,6 +151,31 @@ namespace quda
     }
   }
 
+  /**
+     @brief This instantiate2 function is used to instantiate the
+     precisions, with double precision always enabled.  This is a
+     temporary addition until we fuse this with the original function
+     above when we enforce C++17
```

Comment:
What is stopping us from just enforcing C++17 now?

Reply:
Nothing stopping us.  I would probably prefer C++17 related cleanup in a separate PR in favor of getting this merged first.

Reply:
> Nothing stopping us. I would probably prefer C++17 related cleanup in a separate PR in favor of getting this merged first.

+1 on making it separate